### PR TITLE
Updated theme service to register default themes at service startup

### PR DESCRIPTION
**Pull request by pozil**
Service will now automatically check for missing default themes at startup and install them if needed.

### CODE REVIEW

**Comment by pozil** [comment]
Impacts SP code, see bonita-engine-sp PR 7

File: services/bonita-theme/bonita-theme-impl/src/main/java/org/bonitasoft/engine/theme/impl/ThemeServiceStartupHelper.java [hunk 1]
```diff
@@ -72,22 +76,37 @@ private File unzipTheme() {
...
+     * Create the default Portal and Mobile themes if they do not already exist else do nothing
+     * @throws IOException
+     * @throws SThemeCreationException
+     * @throws SThemeReadException 
+     */
+    public void createDefaultThemes() throws IOException, SThemeCreationException, SThemeReadException {
+    	// Create default Mobile theme if it does not exist
+    	try {
+ 			themeService.getTheme(SThemeType.MOBILE, true);
+ 		} catch (SThemeNotFoundException e) {
```
Comment (educhastenier) [inline services/bonita-theme/bonita-theme-impl/src/main/java/org/bonitasoft/engine/theme/impl/ThemeServiceStartupHelper.java]: Can't we check the existence of a theme without catching exceptions, which is less efficient?

File: services/bonita-theme/bonita-theme-impl/src/main/java/org/bonitasoft/engine/theme/impl/ThemeServiceImpl.java [hunk 2]
```diff
@@ -296,4 +298,28 @@ private void log(final long objectId, final int sQueriableLogStatus, final SPers
         logBuilder.setActionType(actionType);
     }
 
+	@Override
+	public void start() throws SBonitaException {
```
Comment (baptistemesta) [inline services/bonita-theme/bonita-theme-impl/src/main/java/org/bonitasoft/engine/theme/impl/ThemeServiceImpl.java]: Unit test to add on that

**Comment by baptistemesta** [comment]
It's a nice PR but it does not update outdated themes.
It should check if the theme in the classpath is the same as the theme in database juste like we do for profiles.
If we don't do that we will still require migration.
However we can do that in an other PR

**Comment by pozil** [comment]
@educhastenier correct but I just took the legacy code from PlatformAPIImplDelegate here in order to minimize impact. I could rewrite it if needed.

@baptistemesta  you are right but migration is not in my scope, I am targeting upgrade here.

**Comment by baptistemesta** [comment]
Can you redo the pull request on MasterBugFixing branch? We will merge it there to have the ci run before it ends up in master.
Thank you.

**Comment by pozil** [comment]
Done [here](https://github.com/bonitasoft/bonita-engine/pull/46).

**Comment by baptistemesta** [comment]
close this one, will merge https://github.com/bonitasoft/bonita-engine/pull/46

**Pull request closed without merging**